### PR TITLE
TI-1833: Client config fetched from config service

### PR DESCRIPTION
**Pull request by FritzOnFire**
Please review, and advise if I should rewrite my method to completely replace serveGzippedFile or leave it as is.

### CODE REVIEW

File: http.go [hunk 1]
```diff
@@ -286,6 +287,64 @@ type cacheEntry struct {
...
+		w.Write(content)
+		return
+	}
+
+	cacheMutex.RLock()
+	entry, exist := cache[r.URL.Path]
+	cacheMutex.RUnlock()
+
+	// Refresh cache entry if it does not exist or is outdated
+	outdated := exist && (time.Now().Sub(entry.ModTime)/time.Second > 10)
```
Comment (RoanBrand) [inline http.go]: I think `time.Now().Sub(entry.ModTime) > 10*time.Second` is more idiomatic and faster

Comment (bmharper) [inline http.go]: Avoid division. It's harder for CPUs to compute. I think for humans too.
```go
outdated := exist && time.Now().Sub(entry.ModTime) > time.Second*10
```

File: http.go [hunk 2]
```diff
@@ -286,6 +287,64 @@ type cacheEntry struct {
...
+	if !exist || outdated {
+		startGenCache := time.Now()
+		content, err := serviceconfig.GetConfigJson(fpath, serviceName, serviceConfigVersion, serviceConfigFileName)
+		if err != nil {
+			http.NotFound(w, r)
+			return
+		}
+
+		// Detect mime-type
```
Comment (bmharper) [inline http.go]: I think this compression block of code should be factored out into a common function

**Comment by JacoBriers** [comment]
@ryan-vd-merwe does the config service gzip and cache it's responses?

**Comment by bmharper** [comment]
@JacoBriers  Good point

**Comment by bmharper** [comment]
@JacoBriers @ryan-vd-merwe Most of our config should not be accessible from the outside world, as it contains passwords etc, but maybe we can designate certain configs as being public.

**Pull request closed without merging**